### ISSUE TITLE
训练接阶段f1等值均为0

### ISSUE THREAD

**Issue by xiangwf** [macanv/BERT-BiLSTM-CRF-NER#153]
作者大大，您好！我看了很多Issues，都是预测的时候出现f1,precision等为0的情况，而我却总是在训练中这些值为0，求问一下原因
![image](https://user-images.githubusercontent.com/18586019/59155250-b9101d00-8ab7-11e9-946c-245a39f1ffda.png)
我的运行参数是：bert-base-ner-train -data_dir=/data/xiang/ccks-NER -output_dir /data/xiang/ccks-NER/output/ -do_train=True -do_eval=True -do_predict=True -vocab_file=chinese_L-12_H-768_A-12/vocab.txt -bert_config_file=chinese_L-12_H-768_A-12/bert_config.json -init_checkpoint=chinese_L-12_H-768_A-12/bert_model.ckpt -max_seq_length=32 -batch_size=64 -learning_rate=1e-3 -num_train_epochs=3.0 -device_map=0,1 -lstm_size=32
数据集是中文的问答数据，一共有27000多个问句，我试了对实体进行0/1标注以及BIEO标注（并手动改变了bert_lstm_ner.py中labels列表的值），结果都没变，和上面的一样，标签全是0(或O),最后的输出结果也没变过，连修改lstm_size,batch_size,epochs也输出也没变过
是不是我的数据太少的缘故呢？具体怎么改呢？希望您看到后能告诉我一下原因

**Comment by xiangwf** [comment]
已经解决

**Comment by lvjianjin** [comment]
咋解决的呀  我也遇到了

**Comment by cwj** [comment]
晕，我也遇到了

**Comment by cwj** [comment]
> 已经解决

请问怎么解决的？

**Comment by xiangwf** [comment]
> > 已经解决
> 
> 请问怎么解决的？

我的是数据太少造成的

**Comment by xiangwf** [comment]
> 晕，我也遇到了

我的是数据太少造成的，不知道你们的是不是数据的原因

**Comment by xiangwf** [comment]
> 咋解决的呀 我也遇到了

我的是数据太少造成的

**Comment by cwj** [comment]
> > > 已经解决
> > 
> > 
> > 请问怎么解决的？
> 
> 我的是数据太少造成的

请问你多少数据呢，我训练数据99221行，每行平均不到300个字。

**Comment by xiangwf** [comment]
粗糙感觉下来，训练数据起码的有6，7M吧




------------------ 原始邮件 ------------------
发件人: "cwj"<notifications@github.com>;
发送时间: 2019年6月18日(星期二) 下午4:51
收件人: "macanv/BERT-BiLSTM-CRF-NER"<BERT-BiLSTM-CRF-NER@noreply.github.com>;
抄送: "林胥"<1260492827@qq.com>;"State change"<state_change@noreply.github.com>;
主题: Re: [macanv/BERT-BiLSTM-CRF-NER] 训练接阶段f1等值均为0 (#153)



   
已经解决
  
请问怎么解决的？
  
我的是数据太少造成的
  
请问你多少数据呢，我训练数据99221行，每行平均不到300个字。

**Comment by cwj** [comment]
> 粗糙感觉下来，训练数据起码的有6，7M吧
> […](#)
> ------------------ 原始邮件 ------------------ 发件人: "cwj"<notifications@github.com>; 发送时间: 2019年6月18日(星期二) 下午4:51 收件人: "macanv/BERT-BiLSTM-CRF-NER"<BERT-BiLSTM-CRF-NER@noreply.github.com>; 抄送: "林胥"<1260492827@qq.com>;"State change"<state_change@noreply.github.com>; 主题: Re: [macanv/BERT-BiLSTM-CRF-NER] 训练接阶段f1等值均为0 (#153) 已经解决 请问怎么解决的？ 我的是数据太少造成的 请问你多少数据呢，我训练数据99221行，每行平均不到300个字。 — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or mute the thread.

呃，我这边有16M的训练数据，不知道是不是数据量少了的问题，我再加点数据试试，谢谢了哦！

**Comment by xiangwf** [comment]
> > 粗糙感觉下来，训练数据起码的有6，7M吧
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: "cwj"[notifications@github.com](mailto:notifications@github.com); 发送时间: 2019年6月18日(星期二) 下午4:51 收件人: "macanv/BERT-BiLSTM-CRF-NER"[BERT-BiLSTM-CRF-NER@noreply.github.com](mailto:BERT-BiLSTM-CRF-NER@noreply.github.com); 抄送: "林胥"[1260492827@qq.com](mailto:1260492827@qq.com);"State change"[state_change@noreply.github.com](mailto:state_change@noreply.github.com); 主题: Re: [macanv/BERT-BiLSTM-CRF-NER] 训练接阶段f1等值均为0 (#153) 已经解决 请问怎么解决的？ 我的是数据太少造成的 请问你多少数据呢，我训练数据99221行，每行平均不到300个字。 — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 呃，我这边有16M的训练数据，不知道是不是数据量少了的问题，我再加点数据试试，谢谢了哦！

个人建议哦，你还得注意一下数据格式的问题哦！另一个问题是，你的标注得对的上代码中的标注哦，不然得改

**Comment by cwj** [comment]
> > > 粗糙感觉下来，训练数据起码的有6，7M吧
> > > […](#)
> > > ------------------ 原始邮件 ------------------ 发件人: "cwj"[notifications@github.com](mailto:notifications@github.com); 发送时间: 2019年6月18日(星期二) 下午4:51 收件人: "macanv/BERT-BiLSTM-CRF-NER"[BERT-BiLSTM-CRF-NER@noreply.github.com](mailto:BERT-BiLSTM-CRF-NER@noreply.github.com); 抄送: "林胥"[1260492827@qq.com](mailto:1260492827@qq.com);"State change"[state_change@noreply.github.com](mailto:state_change@noreply.github.com); 主题: Re: [macanv/BERT-BiLSTM-CRF-NER] 训练接阶段f1等值均为0 (#153) 已经解决 请问怎么解决的？ 我的是数据太少造成的 请问你多少数据呢，我训练数据99221行，每行平均不到300个字。 — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or mute the thread.
> > 
> > 
> > 呃，我这边有16M的训练数据，不知道是不是数据量少了的问题，我再加点数据试试，谢谢了哦！
> 
> 个人建议哦，你还得注意一下数据格式的问题哦！另一个问题是，你的标注得对的上代码中的标注哦，不然得改

嗯，我的训练数据格式如下：
![image](https://user-images.githubusercontent.com/2813801/59668647-718e3d00-91eb-11e9-95e0-bc0757961bb5.png)
然后我修改了NerProcessor类中get_labels里的一行代码
self.labels = ["O", 'B-TECH', 'I-TECH', "E-TECH", "B-APP", "I-APP", "E-APP", "B-PRO", "I-PRO", "E-PRO", "X", "[CLS]", "[SEP]"]
 这里标签和我标注是一致的
不知道是否还有其它需要修改的地方呢？谢谢啦！

**Comment by xiangwf** [comment]
> > > > 粗糙感觉下来，训练数据起码的有6，7M吧
> > > > […](#)
> > > > ------------------ 原始邮件 ------------------ 发件人: "cwj"[notifications@github.com](mailto:notifications@github.com); 发送时间: 2019年6月18日(星期二) 下午4:51 收件人: "macanv/BERT-BiLSTM-CRF-NER"[BERT-BiLSTM-CRF-NER@noreply.github.com](mailto:BERT-BiLSTM-CRF-NER@noreply.github.com); 抄送: "林胥"[1260492827@qq.com](mailto:1260492827@qq.com);"State change"[state_change@noreply.github.com](mailto:state_change@noreply.github.com); 主题: Re: [macanv/BERT-BiLSTM-CRF-NER] 训练接阶段f1等值均为0 (#153) 已经解决 请问怎么解决的？ 我的是数据太少造成的 请问你多少数据呢，我训练数据99221行，每行平均不到300个字。 — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or mute the thread.
> > > 
> > > 
> > > 呃，我这边有16M的训练数据，不知道是不是数据量少了的问题，我再加点数据试试，谢谢了哦！
> > 
> > 
> > 个人建议哦，你还得注意一下数据格式的问题哦！另一个问题是，你的标注得对的上代码中的标注哦，不然得改
> 
> 嗯，我的训练数据格式如下：
> ![image](https://user-images.githubusercontent.com/2813801/59668647-718e3d00-91eb-11e9-95e0-bc0757961bb5.png)
> 然后我修改了NerProcessor类中get_labels里的一行代码
> self.labels = ["O", 'B-TECH', 'I-TECH', "E-TECH", "B-APP", "I-APP", "E-APP", "B-PRO", "I-PRO", "E-PRO", "X", "[CLS]", "[SEP]"]
> 这里标签和我标注是一致的
> 不知道是否还有其它需要修改的地方呢？谢谢啦！

主要就是这里吧，我也没有很认真的学习代码！话说作者这个代码真的很赞！

**Comment by cwj** [comment]
> > > > > 粗糙感觉下来，训练数据起码的有6，7M吧
> > > > > […](#)
> > > > > ------------------ 原始邮件 ------------------ 发件人: "cwj"[notifications@github.com](mailto:notifications@github.com); 发送时间: 2019年6月18日(星期二) 下午4:51 收件人: "macanv/BERT-BiLSTM-CRF-NER"[BERT-BiLSTM-CRF-NER@noreply.github.com](mailto:BERT-BiLSTM-CRF-NER@noreply.github.com); 抄送: "林胥"[1260492827@qq.com](mailto:1260492827@qq.com);"State change"[state_change@noreply.github.com](mailto:state_change@noreply.github.com); 主题: Re: [macanv/BERT-BiLSTM-CRF-NER] 训练接阶段f1等值均为0 (#153) 已经解决 请问怎么解决的？ 我的是数据太少造成的 请问你多少数据呢，我训练数据99221行，每行平均不到300个字。 — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or mute the thread.
> > > > 
> > > > 
> > > > 呃，我这边有16M的训练数据，不知道是不是数据量少了的问题，我再加点数据试试，谢谢了哦！
> > > 
> > > 
> > > 个人建议哦，你还得注意一下数据格式的问题哦！另一个问题是，你的标注得对的上代码中的标注哦，不然得改
> > 
> > 
> > 嗯，我的训练数据格式如下：
> > ![image](https://user-images.githubusercontent.com/2813801/59668647-718e3d00-91eb-11e9-95e0-bc0757961bb5.png)
> > 然后我修改了NerProcessor类中get_labels里的一行代码
> > self.labels = ["O", 'B-TECH', 'I-TECH', "E-TECH", "B-APP", "I-APP", "E-APP", "B-PRO", "I-PRO", "E-PRO", "X", "[CLS]", "[SEP]"]
> > 这里标签和我标注是一致的
> > 不知道是否还有其它需要修改的地方呢？谢谢啦！
> 
> 主要就是这里吧，我也没有很认真的学习代码！话说作者这个代码真的很赞！

嗯嗯，十分感谢，我再去试试。

**Comment by xiangwf** [comment]
> > > > > > 粗糙感觉下来，训练数据起码的有6，7M吧
> > > > > > […](#)
> > > > > > ------------------ 原始邮件 ------------------ 发件人: "cwj"[notifications@github.com](mailto:notifications@github.com); 发送时间: 2019年6月18日(星期二) 下午4:51 收件人: "macanv/BERT-BiLSTM-CRF-NER"[BERT-BiLSTM-CRF-NER@noreply.github.com](mailto:BERT-BiLSTM-CRF-NER@noreply.github.com); 抄送: "林胥"[1260492827@qq.com](mailto:1260492827@qq.com);"State change"[state_change@noreply.github.com](mailto:state_change@noreply.github.com); 主题: Re: [macanv/BERT-BiLSTM-CRF-NER] 训练接阶段f1等值均为0 (#153) 已经解决 请问怎么解决的？ 我的是数据太少造成的 请问你多少数据呢，我训练数据99221行，每行平均不到300个字。 — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or mute the thread.
> > > > > 
> > > > > 
> > > > > 呃，我这边有16M的训练数据，不知道是不是数据量少了的问题，我再加点数据试试，谢谢了哦！
> > > > 
> > > > 
> > > > 个人建议哦，你还得注意一下数据格式的问题哦！另一个问题是，你的标注得对的上代码中的标注哦，不然得改
> > > 
> > > 
> > > 嗯，我的训练数据格式如下：
> > > ![image](https://user-images.githubusercontent.com/2813801/59668647-718e3d00-91eb-11e9-95e0-bc0757961bb5.png)
> > > 然后我修改了NerProcessor类中get_labels里的一行代码
> > > self.labels = ["O", 'B-TECH', 'I-TECH', "E-TECH", "B-APP", "I-APP", "E-APP", "B-PRO", "I-PRO", "E-PRO", "X", "[CLS]", "[SEP]"]
> > > 这里标签和我标注是一致的
> > > 不知道是否还有其它需要修改的地方呢？谢谢啦！
> > 
> > 
> > 主要就是这里吧，我也没有很认真的学习代码！话说作者这个代码真的很赞！
> 
> 嗯嗯，十分感谢，我再去试试。

嗯嗯，加油

**Comment by rick29sz** [comment]
"X"是什么含义呢？？
